### PR TITLE
Backend changes to support markdown in the frontend

### PR DESCRIPTION
**Pull request by DougC**
## Do not merge!
Please don't merge this PR yet! I've opened it so we can have a discussion about whether we think this is the way we'd like to go, so any and all comments welcome.

This branch changes the way the section text is held. Rather than a separate paragraph table, the text is held in a column on the `section` table (up to 4k) and send as a string to the frontend. I've included evolution files that create the column and populate the existing sections with the same text that was previously held in the `paragraph` table. One nice side-effect is that it removes a lot of the complexity associated with joining three tables and building up the opportunity structure.

### CODE REVIEW

**Comment by DougC** [comment]
So one implication of this is on how we count words. For example, if the user wants to create a list then they'll do something like:
```
* one
* two
* three
```
but we probably don't want to count the `*`s as words. That's pretty easy to deal with, but what if they want a numbered list:
```
1. one
2. two
3. three
```
then how do we know that these numbers are part of the list-making markdown and not count them, but still count other numbers in the text (if, indeed, we want to get that specific about things)?

File: src/main/resources/evolutions/default/12.sql [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+This may be by sharing knowledge, commercialising ideas, exploring social benefits or other ways to increase the impact of your research.
+
+Under the Exploring Innovation Seminars programme, we will pay up to £2,000 for each event promoting innovation and collaboration. We will not pay for food or drink.
+
+Only organisations which receive funding from UK Research Councils may apply.
+
+As part of this, we want to help you develop innovative ways of building on the research you carry out.');
+INSERT INTO section (id, section_number, opportunity_id, title, text) VALUES (2, 2, 1, 'What we will ask you', '### Event title
+What is your event called? Wordcount: 20
```
Comment (jon-shipley) [inline src/main/resources/evolutions/default/12.sql]: `Wordcount: 20` should be on a seperate line - this is supported in Markdown by using: <space> <space> <return> in the line above.

Comment (jon-shipley) [inline src/main/resources/evolutions/default/12.sql]: This seems excellent to me.    I saw that https://github.com/alphagov/govspeak have something similar where it is customised to support some .gov.uk styles.  Not scala though.

We would need to do some work here to get it to output the "right" html/css as at the moment it doesn't quite produce the right thing.

Comment (DougC) [inline src/main/resources/evolutions/default/12.sql]: Yeah. Laika has some good support for customisation of the html. See http://planet42.github.io/Laika/customizing-laika/customize-rendering.html for some examples.

**Comment by DougC** [comment]
Closing this (and the frontend PR) after conferring with Jon. Doesn't seem like the right direction to go in at the moment.

**Pull request closed without merging**